java/code/src/com/suse/manager/webui/controllers/VirtualPoolsController.java

```diff
@@ -220,4 +269,68 @@ private static ModelAndView renderPage(Request request, Response response, User
 
         return new ModelAndView(data, String.format("templates/virtualization/pools/%s.jade", template));
     }
+
+    private static String poolAction(Request request, Response response, User user,
+            Function<VirtualPoolBaseActionJson, BaseVirtualizationPoolAction> actionCreator) {
+        return poolAction(request, response, user, actionCreator, VirtualPoolBaseActionJson.class);
+    }
+
+    private static String poolAction(Request request, Response response, User user,
+            Function<VirtualPoolBaseActionJson, BaseVirtualizationPoolAction> actionCreator,
+            Class<? extends VirtualPoolBaseActionJson> jsonClass) {
+        Long serverId;
+
+        try {
+            serverId = Long.parseLong(request.params("sid"));
+        }
+        catch (NumberFormatException e) {
+            throw new NotFoundException();
+        }
+        Server host = SystemManager.lookupByIdAndUser(serverId, user);
+
+        VirtualPoolBaseActionJson data;
+        try {
+            data = GSON.fromJson(request.body(), jsonClass);
+        }
+        catch (JsonParseException e) {
+            throw Spark.halt(HttpStatus.SC_BAD_REQUEST);
+        }
+
+        if (!data.getPoolNames().isEmpty()) {
+            Map<String, String> actionsResults = data.getPoolNames().stream().collect(
+                    Collectors.toMap(Function.identity(),
+                poolName -> {
+                    return scheduleAction(poolName, user, host, actionCreator, data);
+                }
+            ));
+            return json(response, actionsResults);
+        }
+
+        String result = scheduleAction(null, user, host, actionCreator, data);
```

Comment:
I wonder about the behavior of the `pool-refreshed.sls` (and in turn of the `virt.pool_refresh` execution module as well) if the pool name is null. Does it refresh all pools/the default one/...?

Reply:
That should not happen since the UI prevents that. But if that really happens, salt will happily pass it to libvirt which in turn will raise an exception.

Reply:
In this case, why do we need this code here? Is this for other types of pool actions (in other PRs)?

Reply:
yes it's for the other types of pool actions